### PR TITLE
Cleanup unused entitlements functions, add comments

### PR DESCRIPTION
**Pull request by clemire**

### CODE REVIEW

File: packages/web3/src/ConvertersEntitlements.ts [hunk 1]
```diff
@@ -1,25 +1,11 @@
 import { ethers } from 'ethers'
 
 import { Address, EntitlementStruct } from './ContractTypes'
-import { Hex, decodeAbiParameters, parseAbiParameters } from 'viem'
 import { encodeEntitlementData } from './entitlement'
 import { IRuleEntitlementBase } from './v3'
 
 const UserAddressesEncoding = 'address[]'
 
-export function decodeRuleData(encodedData: string): string[] {
```
Comment (clemire) [inline packages/web3/src/ConvertersEntitlements.ts:10]: Not used and also incorrect - uses UserEntitlement data type below.

File: packages/web3/src/entitlement.ts [hunk 2]
```diff
@@ -18,6 +10,12 @@ import { MOCK_ADDRESS } from './Utils'
 
 const zeroAddress = ethers.constants.AddressZero
 
+// This public client is used purely to infer the typescript type of RuleData.
```
Comment (clemire) [inline packages/web3/src/entitlement.ts:13]: @texuf if you know a better way here? I took a quick look at ethers docs but nothing jumped out.

Comment (texuf) [reply packages/web3/src/entitlement.ts:13]: Can you try the ExtractAbi function that @shuhuiluo used here: https://github.com/river-build/river/pull/675/files

Comment (clemire) [reply packages/web3/src/entitlement.ts]: We actually don't need the function, just the type here. I found another approach, PTAL?

File: packages/web3/src/entitlement.ts [hunk 3]
```diff
@@ -213,22 +179,6 @@ export function encodeEntitlementData(ruleData: IRuleEntitlementBase.RuleDataStr
...
-        })
-
-    if (!getRuleDataAbi) {
-        throw new Error('getRuleData ABI not found')
-    }
-    return decodeAbiParameters(
-        getRuleDataAbi.outputs,
-        entitlementData,
-    ) as unknown as IRuleEntitlementBase.RuleDataStruct[]
-}
```
Comment (texuf) [inline packages/web3/src/entitlement.ts:230]: looks like this one is used in tests ^

Comment (clemire) [reply packages/web3/src/entitlement.ts:230]: added back in

**Review by texuf — APPROVED** [review]

**Pull request merged**